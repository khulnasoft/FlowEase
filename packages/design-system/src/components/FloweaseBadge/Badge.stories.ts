import type { StoryFn } from '@storybook/vue3';
import FloweaseBadge from './Badge.vue';

export default {
	title: 'Atoms/Badge',
	component: FloweaseBadge,
	argTypes: {
		theme: {
			type: 'text',
			options: ['default', 'primary', 'secondary', 'tertiary'],
		},
		size: {
			type: 'select',
			options: ['small', 'medium', 'large'],
		},
	},
};

const Template: StoryFn = (args, { argTypes }) => ({
	setup: () => ({ args }),
	props: Object.keys(argTypes),
	components: {
		FloweaseBadge,
	},
	template: '<flowease-badge v-bind="args">Badge</flowease-badge>',
});

export const Badge = Template.bind({});
Badge.args = {};
