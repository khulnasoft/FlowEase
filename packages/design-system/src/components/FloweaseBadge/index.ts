import FloweaseBadge from './Badge.vue';

export default FloweaseBadge;
