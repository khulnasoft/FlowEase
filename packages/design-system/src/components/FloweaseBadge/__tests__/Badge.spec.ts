import { render } from '@testing-library/vue';
import FloweaseBadge from '../Badge.vue';

describe('components', () => {
	describe('FloweaseBadge', () => {
		describe('props', () => {
			it('should render default theme correctly', () => {
				const wrapper = render(FloweaseBadge, {
					props: {
						theme: 'default',
						size: 'large',
						bold: true,
					},
					slots: {
						default: '<flowease-text>Default badge</flowease-text>',
					},
					global: {
						stubs: ['flowease-text'],
					},
				});
				expect(wrapper.html()).toMatchSnapshot();
			});
			it('should render secondary theme correctly', () => {
				const wrapper = render(FloweaseBadge, {
					props: {
						theme: 'secondary',
						size: 'medium',
						bold: false,
					},
					slots: {
						default: '<flowease-text>Secondary badge</flowease-text>',
					},
					global: {
						stubs: ['flowease-text'],
					},
				});
				expect(wrapper.html()).toMatchSnapshot();
			});
			it('should render with default values correctly', () => {
				const wrapper = render(FloweaseBadge, {
					slots: {
						default: '<flowease-text>A Badge</flowease-text>',
					},
					global: {
						stubs: ['flowease-text'],
					},
				});
				expect(wrapper.html()).toMatchSnapshot();
			});
		});
	});
});
