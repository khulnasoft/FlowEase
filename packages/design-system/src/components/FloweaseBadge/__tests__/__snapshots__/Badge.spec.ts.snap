// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components > FloweaseBadge > props > should render default theme correctly 1`] = `"<span class="flowease-badge default"><flowease-text-stub bold="true" size="large" compact="true" tag="span"></flowease-text-stub></span>"`;

exports[`components > FloweaseBadge > props > should render secondary theme correctly 1`] = `"<span class="flowease-badge secondary"><flowease-text-stub bold="false" size="medium" compact="true" tag="span"></flowease-text-stub></span>"`;

exports[`components > FloweaseBadge > props > should render with default values correctly 1`] = `"<span class="flowease-badge default"><flowease-text-stub bold="false" size="small" compact="true" tag="span"></flowease-text-stub></span>"`;
